activesupport/lib/active_support/cache.rb

```diff
@@ -796,9 +796,11 @@ def normalize_options(options)
         # Expands and namespaces the cache key.
         # Raises an exception when the key is +nil+ or an empty string.
         # May be overridden by cache stores to do additional normalization.
-        def normalize_key(key, options = nil)
+        def normalize_key(key, options = nil, validate: true)
           str_key = expanded_key(key)
-          raise(ArgumentError, "key cannot be blank") if !str_key || str_key.empty?
+          if validate && (!str_key || str_key.empty?)
```

Comment:
While we're here, could this use `blank?`:

```suggestion
          if validate && str_key.blank?
```

Reply:
I don't think so, it breaks some instances (there's tests for these) where various string encodings are used for the key which would raise here: https://github.com/rails/rails/blob/main/activesupport/lib/active_support/core_ext/object/blank.rb#L128

Reply:
That sounds like a bug in blank? to me 🤔

Reply:
Do you have a link to the tests?

Reply:
Oh actually, that may have been my attempt to use `blank?` before the key was expanded...
I'll update and hopefully nothing should break!

Reply:
I feel like `String#blank?` should not raise regardless of input, but that's separate to this PR.